crates/starknet-os/src/hints/kzg.rs

```diff
@@ -0,0 +1,285 @@
+use std::collections::HashMap;
+use std::io::{self, Read};
+use std::path::Path;
+
+use c_kzg::{Blob, KzgCommitment, BYTES_PER_FIELD_ELEMENT};
+use cairo_vm::hint_processor::builtin_hint_processor::hint_utils::{
+    get_ptr_from_var_name, get_relocatable_from_var_name,
+};
+use cairo_vm::hint_processor::hint_processor_definition::HintReference;
+use cairo_vm::serde::deserialize_program::ApTracking;
+use cairo_vm::types::exec_scope::ExecutionScopes;
+use cairo_vm::types::relocatable::MaybeRelocatable;
+use cairo_vm::vm::errors::hint_errors::HintError;
+use cairo_vm::vm::vm_core::VirtualMachine;
+use cairo_vm::Felt252;
+use indoc::indoc;
+use num_bigint::BigInt;
+use num_traits::{Num, One, Zero};
+
+use crate::utils::execute_coroutine;
+
+const FIELD_ELEMENTS_PER_BLOB: usize = 4096;
+const COMMITMENT_BYTES_LENGTH: usize = 48;
+const COMMITMENT_LOW_BIT_LENGTH: usize = (COMMITMENT_BYTES_LENGTH * 8) / 2;
+const BLOB_SUBGROUP_GENERATOR: &str = "39033254847818212395286706435128746857159659164139250548781411570340225835782";
+const BLS_PRIME: &str = "52435875175126190479447740508185965837690552500527637822603658699938581184513";
+
+fn is_power_of_2(n: usize) -> bool {
+    n > 0 && (n & (n - 1)) == 0
+}
+
+fn fft(coeffs: &[BigInt], generator: &BigInt, prime: &BigInt, bit_reversed: bool) -> Result<Vec<BigInt>, io::Error> {
```

Comment:
What is the original source of this code?

Reply:
Would be good to leave some kind of hint to help others find this. In other places we've copied the code comments from the fn (just below the line you linked to), which helps when `grep`ing.